File: lualearn/circulo.lua
function CalcularAreaCirculo(radio)
	return math.pi * radio ^ 2
end

io.write("Ingrese el radio del circulo: ")
local radio = tonumber(io.read())

if radio and radio > 0 then
	local area = CalcularAreaCirculo(radio)
	print("El area del circulo es: " .. area)
else
	print("El radio debe ser un numero positivo")
end

File: Java/POO/src/main/java/FiguraGeometrica/Circulo.java
package FiguraGeometrica;

public class Circulo extends FiguraGeometrica {
    private double radio;

    public Circulo(String nombre, double radio) {
        super(nombre);

        if (radio > 0) {
            this.radio = radio;
        } else {
            this.radio = 0;
            System.out.println("El radio debe ser positivo");
        }
    }

    public double getRadio() {
        return radio;
    }

    @Override
    public double calcularArea() {
        return Math.PI * radio * radio;
    }
}
